doc/dev.md

```diff
@@ -0,0 +1,158 @@
+# Development guide
+
+## Getting the source code
+
+First check out the source code.
+The following commands will fetch the latest code and also make
+sure that submodules are properly initialized.
+
+```sh
+# For this document we will assume you are inside a folder for development.
+# We will refer to that as `dev`.
+#
+# I usually have a folder called `dev` in my user folder and inside a subfolder
+# for each software project, here that would be `/home/user/dev/qfield`.
+# You are free to choose yours.
+
+git clone git@github.com:opengisch/QField.git
+git submodule update --init --recursive
+# Alternatively you can ues the following URL in case you have not set up SSH keys for github
+#   https://github.com/opengisch/QField.git
+```
+
+## Linux
+
+You need to have cmake installed.
+
+You have two options to build QField. Using system packages which will
+reuse packages installed from your package manager. Or using vcpkg which
+will build all the packages from source.
+
+### Using system packages
+
+This will use your system packages.
+Make soure you have installed the appropriate `-dev` or `-devel` packages
```

Comment:
```suggestion
Make sure you have installed the appropriate `-dev` or `-devel` packages
```

What is more, what are "the appropriate -dev or -develo packages"? Do we have the list somewhere? An `apt install` or `yum install` with the list is very, very handy, so it's easy to translate in your own linux flavor.

Reply:
I would have been more precise if I would know, I think this will be assembled over time with experience.